packages/flutter_tools/lib/src/macos/xcdevice.dart

```diff
@@ -564,6 +561,14 @@ class XCDevice {
           if (code != -10) {
             isConnected = false;
           }
+          // Error: iPhone is not paired with your computer. To use iPhone with Xcode, unlock it and choose to trust this computer when prompted. (code -9)
+          if (errorMessage != null && errorMessage.contains('not paired')) {
+            UsageEvent('device', 'ios-trust-failure', flutterUsage: globals.flutterUsage).send();
+            _analytics.send(Event.appleUsageEvent(workflow: 'device', parameter: 'ios-trust-failure'));
+            if (code == -9) {
```

Comment:
I can't remember if there's a case where we should post the usage event when the code isn't -9, so I kept the old behavior of always posting it if the error message contains "not paired"